uwblueprint/community-fridge-kw
# fix: recurring donation dates

## Brief description. What is this change? 

### [One Less Schedule Created](https://www.notion.so/uwblueprintexecs/One-less-schedule-created-for-recurring-donations-after-7pm-507579bce3fc472b86dac5ed86a607d9)



## Implementation description. How did you make this change?
* Changed recurring donation end date to be set as end of day in the frontend. Currently recurring end date is set to be the date that is selected in the date picker and the time is just the current time, but with the change I made, time is now end of day. The old timing messed up how donations are created in the backend and sometimes one less schedule was created because of this. Take a look at the following screenshot: 
<img width="1881" alt="image" src="https://user-images.githubusercontent.com/37384882/159151665-0fd6f558-c8e7-4a24-a1de-a32f7c5d3f07.png">
A recurring donation should have been added for Mar 26 but it did not.

Now you might be thinking why could I not do this change in the backend function instead of frontend?
This is because when we use dayjs in the backend, it converts the time into UTC. This creates some edge cases bugs, for example lets say recurring donation end date was Mar 27 10pm, with dayjs in backend this will become Mar 28 2am (since UTC is 4 hours ahead of EST), then if I set the date to EOD as I did in frontend right now, recurring end date will become Mar 28 11:59 am which will then create an extra donation cuz of how our while loop and recurring donation logic is implemented.

Why doesnt the UTC conversion happen in the frontend?
From my understanding, in the `SelectDateTime.tsx`, we are using Javascript dates and timezone is preserved so it is not converted to UTC

Lol this is very hard to explain in text, but if someone is confused with this, lmk and I can explain in more depth but I am hoping this can fix the recurring donation dates saga once and for all (fingers crossed).



## Steps to test
1. Create different daily recurring donations with different times and edge cases (4am, 12am, 8pm, etc) and ensure accurate number of daily donations are being created. If end date is Mar 26 for example, the last donation for that should be on Mar 26 as well.




## Checklist
